### DISCUSSION TITLE
Callback Question in C++ Project

### DISCUSSION THREAD

**Discussion by roryy19** [slint-ui/slint#6757] [Q&A]
Here is a mock situation of my problem:

- Component A is a slint file that has a callback.
- Component B is another slint file which inherits Component A.
- main.cpp calls Component B.

Question:
How does main.cpp access callbacks from Component A, which it does not have direct access to, since main.cpp only inherits Component B?

**Comment by ogoffart** [comment]
Two ways:
 1. forward the callback as an explicit callback in your component B
 2. use a global (which you can access in C++ with the `global` generated function https://docs.slint.dev/latest/docs/cpp/generated_code#global-singletons )

See also https://github.com/slint-ui/slint/discussions/6733#discussioncomment-11174982
